src/unix/linux.c

```diff
@@ -1627,53 +1627,34 @@ uint64_t uv__hrtime(uv_clocktype_t type) {
 }
 
 
-int uv_resident_set_memory(size_t* rss) {
+int uv_resident_set_memory(size_t* rss)
+{
   char buf[1024];
   const char* s;
-  ssize_t n;
   long val;
-  int fd;
+  int rc;
   int i;
 
-  do
-    fd = open("/proc/self/stat", O_RDONLY);
-  while (fd == -1 && errno == EINTR);
-
-  if (fd == -1)
-    return UV__ERR(errno);
-
-  do
-    n = read(fd, buf, sizeof(buf) - 1);
-  while (n == -1 && errno == EINTR);
-
-  uv__close(fd);
-  if (n == -1)
-    return UV__ERR(errno);
-  buf[n] = '\0';
-
-  s = strchr(buf, ' ');
-  if (s == NULL)
-    goto err;
-
-  s += 1;
-  if (*s != '(')
-    goto err;
+  /* rss: 24th element */
+  rc = uv__slurp("/proc/self/stat", buf, sizeof(buf));
+  if (rc < 0)
+    return rc;
 
-  s = strchr(s, ')');
+  /* find the last ')' */
+  s = strrchr(buf, ')');
   if (s == NULL)
     goto err;
 
-  for (i = 1; i <= 22; i++) {
+  for (i = 1; i <= 22; i++)
+  {
     s = strchr(s + 1, ' ');
     if (s == NULL)
       goto err;
   }
 
   errno = 0;
   val = strtol(s, NULL, 10);
-  if (errno != 0)
-    goto err;
-  if (val < 0)
+  if (val < 0 || errno != 0)
```

Comment:
I can't post this as a diff but please undo. Stylistic and functional changes shouldn't be intermixed in a single commit.

Reply:
I separated the commit and reopened a new PR